apps/media/src/app/app.controller.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  Controller,
+  Delete,
+  Get,
+  Param,
+  ParseFilePipe,
+  Post,
+  Query,
+  Res,
+  UploadedFile,
+  UseGuards,
+  UseInterceptors
+} from '@nestjs/common'
+import {FileInterceptor} from '@nestjs/platform-express'
+import {TokenAuthGuard} from '@wepublish/media/api'
+import {Response} from 'express'
+import 'multer'
+import {v4 as uuidv4} from 'uuid'
+import {AppService} from './app.service'
+import {SupportedImagesValidator} from './supported-images-validator'
+import {TransformationsDto} from './transformations.dto'
+
+@Controller({
+  version: '1'
+})
+export class AppController {
+  constructor(private readonly appService: AppService) {}
+
+  @UseGuards(TokenAuthGuard)
+  @Post()
+  @UseInterceptors(FileInterceptor('file'))
+  async uploadImage(
+    @Res() res: Response,
+    @UploadedFile(
+      new ParseFilePipe({
+        fileIsRequired: true,
+        validators: [new SupportedImagesValidator()]
+      })
+    )
+    uploadedFile: Express.Multer.File
+  ) {
+    const imageId = uuidv4()
+    const metadata = await this.appService.saveImage(imageId, uploadedFile.buffer)
+
+    res.status(201).send({
+      id: imageId,
+      filename: `${imageId}.webp`,
```

Comment:
if we want to use webp for now, but move to avif in the future, maybe the value could be parametrized to make it easier and less error prone to switch?

Reply:
Avif will be the output for later, I think it makes sense to have everything in webp (or avif to begin with) and only modify the what we return. This is for the DB entry that shows the filename. Not sure we even need this in the editor and maybe can drop it?